virttest/libvirt_xml/vm_xml.py

```diff
@@ -819,15 +820,26 @@ def set_vm_vcpus(vm_name, vcpus, current=None, sockets=None, cores=None,
                 vmxml['cpu'] = vmcpu_xml
             try:
                 vmcpu_xml = vmxml['cpu']
-                if update_numa and vmxml.cpu.numa_cell:
-                    no_numa_cell = len(vmxml.cpu.numa_cell)
+                if (update_numa and vmxml.cpu.numa_cell) or numa_number is not None:
```

Comment:
I assume you can separate out numa_number is not None  as another if statement ,rather than mix together to make code confused

Reply:
Updated.